File: app/src/main/java/com/lingoswap/domain/usecase/auth/RegisterUseCase.java
package com.lingoswap.domain.usecase.auth;

import com.lingoswap.data.model.AuthResponse;
import com.lingoswap.data.model.request.RegisterRequest;
import com.lingoswap.data.repository.RepositoryCallback;
import com.lingoswap.domain.repository.AuthRepository;

import javax.inject.Inject;

public class RegisterUseCase {
    private final AuthRepository repository;

    @Inject
    public RegisterUseCase(AuthRepository repository) {
        this.repository = repository;
    }

    public void execute(RegisterRequest request, RepositoryCallback<AuthResponse> callback) {
        repository.register(request, callback);
    }
}

File: app/src/main/java/com/lingoswap/domain/usecase/auth/RegisterUseCase.kt
package com.lingoswap.domain.usecase.auth

import com.lingoswap.data.model.request.RegisterRequest
import com.lingoswap.data.model.response.AuthResponse
import com.lingoswap.domain.repository.AuthRepository
import com.lingoswap.utils.Resource
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class RegisterUseCase @Inject constructor(
    private val repository: AuthRepository
) {
    suspend operator fun invoke(request: RegisterRequest): Flow<Resource<AuthResponse>> {
        return repository.register(request)
    }
}
